lib/spack/spack/installer.py

```diff
@@ -401,13 +402,11 @@ def _try_install_from_binary_cache(pkg, explicit, unsigned=False):
     """
     pkg_id = package_id(pkg)
     tty.debug('Searching for binary cache of {0}'.format(pkg_id))
-    matches = binary_distribution.get_mirrors_for_spec(pkg.spec)
 
-    if not matches:
+    if not spack.mirror.MirrorCollection():
         return False
 
-    return _process_binary_cache_tarball(pkg, pkg.spec, explicit, unsigned,
-                                         mirrors_for_spec=matches)
+    return _process_binary_cache_tarball(pkg, pkg.spec, explicit, unsigned)
```

Comment:
Because this is the only location where `_process_binary_cache_tarball` gets called, it should now be safe to remove the `mirrors_for_spec` parameter to that function.

Reply:
Fair point.  I guess in that case I should remove the `mirrors_for_spec` parameter from `binary_distribution.download_tarball()` then too, since after this no call site of that method is using it anymore.